samples/kotlin-android-app/src/main/java/com/appcues/samples/kotlin/ExampleApplication.kt

```diff
@@ -39,6 +42,21 @@ class ExampleApplication : Application() {
 
         appcues = Appcues(this, BuildConfig.APPCUES_ACCOUNT_ID, BuildConfig.APPCUES_APPLICATION_ID) {
             loggingLevel = if (BuildConfig.DEBUG) LoggingLevel.DEBUG else LoggingLevel.INFO
+            navigationHandler = object : NavigationHandler {
+                // This is an example where we're processing navigation requests coming from Appcues experiences,
+                // but simply forwarding them on to start an Activity from an Intent with the given Uri. If an application
+                // had a more sophisticated deep linking mechanism and needed finer control over reporting the completion
+                // of link navigation - this is where the SDK would allow hooking in and supplying that.
+                override suspend fun navigateTo(uri: Uri): Boolean {
+                    Intent(Intent.ACTION_VIEW).apply {
+                        data = uri
+                        flags = Intent.FLAG_ACTIVITY_NEW_TASK
+                    }.also {
+                        startActivity(it)
+                    }
+                    return true // navigation successful
+                }
+            }
```

Comment:
seems like this would be generally less necessary on Android than on iOS, where the AndroidManifest intent-filter system is more prescriptive about routing paths to Activities. On iOS, it tends to be more custom logic and chaining of ViewController presentations with animations and completion - which the host app needs to inform the SDK about completion.

I'm imagining perhaps a scenario on Android where a more sophisticated deep linking system is actually doing various Fragment reconfigurations - or even a Compose app recomposing itself in a more advanced way. I want to provide some safety hook to allow the host app to control navigation and inform the SDK when done, so we can safely present the next step _after_ navigation completes.